include/deal.II/base/geometry_info.h

```diff
@@ -904,6 +904,85 @@ class RefinementCase : public RefinementPossibilities<dim>
 
 
 
+struct DirectionPossibilites
+{
+  enum Possibilities : std::uint8_t
+  {
+    /**
```

Comment:
@dominiktassilostill and I discussed whether or not this would be the right place for the new infrastructure to detect refinement possibilities. It mirrors somewhat the behavior of the anisotropic refinement, but obviously the longer-term plan would be to eliminate `GeometryInfo`. Thus, we would be happy for alternative suggestions of where to place this and possibly also the naming.

Reply:
I think all of this information should go into `ReferenceCell` because we may encounter similar issues with wedges and pyramids. In fact, the different choice tables that I believe are now in `tria.cc` should probably be in member functions of `ReferenceCell` as well.

For this specific class: It is specific to tetrahedra, but neither name nor member names reflect this. I think a better interface would be `ReferenceCell::n_isotropic_refinement_choices()` which returns how many different ways there are to refine a cell (returning 1 for all cells but tets), and then other member functions that can be used to query things about the child cells depending on the refinement choice.